packages/server/src/fhir/repo.ts

```diff
@@ -521,13 +523,42 @@ export class Repository extends BaseRepository implements FhirRepository<PoolCli
       throw new OperationOutcomeError(forbidden);
     }
 
+    await this.handleBinaryData(result);
     await this.handleMaybeCacheOnly(result);
     await setCacheEntry(result);
     await addBackgroundJobs(result, { interaction: create ? 'create' : 'update' });
     this.removeHiddenFields(result);
     return result;
   }
 
+  /**
+   * Handles a Binary resource with embedded base-64 data.
+   * Writes the data to the binary storage and removes the data field from the resource.
+   * @param resource - The resource to write to the database.
+   */
+  private async handleBinaryData(resource: Resource): Promise<void> {
+    if (resource.resourceType !== 'Binary' || !resource.data) {
+      return;
+    }
+
+    // Parse result.data as a base64 string
+    const buffer = Buffer.from(resource.data, 'base64');
+
+    // Convert buffer to a Readable stream
+    const stream = new Readable({
+      read() {
+        this.push(buffer);
+        this.push(null); // Signifies the end of the stream (EOF)
+      },
+    });
+
+    // Write the stream to the binary storage
+    await getBinaryStorage().writeBinary(resource, undefined, resource.contentType, stream);
+
+    // Remove the data field from the resource
+    delete resource.data;
```

Comment:
```suggestion
    resource.data = undefined;
```
Setting to `undefined` usually preferred imo, just because of complicated hidden class implications. And `undefined` is already allowed in the types 🤔 so should be ok?